### PR TITLE
libFuzzer: add CI fuzzers build, add tcp protocol fuzzer, fix other fuzzers.

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: programs/server/CMakeLists.txt [hunk 1]
```diff
@@ -37,3 +37,5 @@ clickhouse_embed_binaries(
     RESOURCES config.xml users.xml embedded.xml play.html dashboard.html js/uplot.js
 )
 add_dependencies(clickhouse-server-lib clickhouse_server_configs)
+
+add_subdirectory(fuzzers)
```
Comment (evillique) [inline programs/server/CMakeLists.txt:41]: Shouldn't it be under the ENABLE_FUZZING option?
```suggestion
if (ENABLE_FUZZING)
    add_subdirectory(fuzzers)
endif ()
```

File: programs/server/fuzzers/tcp_protocol_fuzzer.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+{
+    for (int i = 1; i < *argc; ++i)
+    {
+        if ((*argv)[i][0] == '-')
+        {
+            if ((*argv)[i][1] == '-')
+                args.push_back((*argv)[i]);
+            else
+            {
+                if (strncmp((*argv)[i], "-host=", 6) == 0)
```
Comment (alexey-milovidov) [inline programs/server/fuzzers/tcp_protocol_fuzzer.cpp:37]: boost::program_options is ok.

**Comment by nikitamikhaylov** [comment]
Hmm, I don't see this fuzzer in the output 
```
+ BINARIES_SIZE='273M	/src/ClickHouse/build/src/Core/fuzzers/names_and_types_fuzzer
12M	/src/ClickHouse/build/src/Parsers/fuzzers/lexer_fuzzer
55M	/src/ClickHouse/build/src/Parsers/fuzzers/codegen_fuzzer
43M	/src/ClickHouse/build/src/Parsers/fuzzers/codegen_fuzzer/codegen_select_fuzzer
29M	/src/ClickHouse/build/src/Parsers/fuzzers/create_parser_fuzzer
29M	/src/ClickHouse/build/src/Parsers/fuzzers/select_parser_fuzzer
21M	/src/ClickHouse/build/src/Storages/fuzzers/mergetree_checksum_fuzzer
273M	/src/ClickHouse/build/src/Storages/fuzzers/columns_description_fuzzer
21M	/src/ClickHouse/build/src/Compression/fuzzers/lz4_decompress_fuzzer
21M	/src/ClickHouse/build/src/Compression/fuzzers/compressed_buffer_fuzzer
21M	/src/ClickHouse/build/src/Compression/fuzzers/delta_decompress_fuzzer
21M	/src/ClickHouse/build/src/Compression/fuzzers/double_delta_decompress_fuzzer
23M	/src/ClickHouse/build/src/Compression/fuzzers/encrypted_decompress_fuzzer'
```

**Comment by evillique** [comment]
It seems like `build.sh` checks out the master branch and not the branch from the PR.

**Comment by nikitamikhaylov** [comment]
But how then it is intended to work in CI? I thought it should checkout repo and use current branch

**Comment by yakov-olkhovskiy** [comment]
Style Check:
```
/ClickHouse/.github/workflows/pull_request.yml
  3524:1    warning  comment not indented like content  (comments-indentation)
```
we have this indentation everywhere - it should not fail...

**Comment by evillique** [comment]
> But how then it is intended to work in CI? I thought it should checkout repo and use current branch

Yep, I have looked up the docs and logs and CIFuzz does indeed checks out on the branch of this PR. The problem was in another part of the `build.sh`. 

This PR should solve this: https://github.com/google/oss-fuzz/pull/8897

**Comment by alexey-milovidov** [comment]
What happened?

**Comment by yakov-olkhovskiy** [comment]
I need to update CI and building stuff here I suppose - there were many changes since... and update corpus...

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit e9ccfb24f14955474ec987900de919f595f2e214 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/42599/e9ccfb24f14955474ec987900de919f595f2e214/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by yakov-olkhovskiy** [comment]
@nikitamikhaylov what was an intention of building and running steps for fuzzers on our CI?

**Comment by nikitamikhaylov** [comment]
Completely green!

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**